port_drayage_plugin/src/port_drayage_worker.cpp

```diff
@@ -117,26 +117,51 @@ namespace port_drayage_plugin
         using boost::property_tree::ptree;
         ptree pt;
         pt.put("cmv_id", _cmv_id);
-        pt.put("operation", PORT_DRAYAGE_ARRIVAL_OPERATION_ID);
 
         // Add current vehicle location (latitude and longitude)
         ptree location;
         location.put("latitude", _current_gps_position.latitude); 
         location.put("longitude", _current_gps_position.longitude); 
         pt.put_child("location", location);
 
-        // Add cargo-related fields based on whether the vehicle is currently carrying cargo
-        if (_has_cargo) {
-            pt.put("cargo", true);
-            pt.put("cargo_id", _cargo_id);
-        }
-        else {
-            pt.put("cargo", false);
+        // Add flag to indicate whether CMV is carring cargo
+        pt.put("cargo", _has_cargo);
+
+        // If CMV has not received any port drayage messages yet, assign 'operation' field for its initial arrival message
+        if (!_has_received_first_mobility_operation_msg) {
```

Comment:
Seems like maybe this kind of value could be moved into the state machine (I see references to a _pdsm in here, which I'm assume is Port Drayage State Machine) as the ultimate keeper of plugin state. And that this should be checking if we're in a state where we need to encode this data more so than the indirect property that we've received a mobility message, if that makes sense.

Reply:
I agree it is much cleaner to update the state machine for this logic. I have made an update to account for this.